llvm/test/tools/llvm-split/SYCL/device-code-split/one-kernel-per-module.ll

```diff
@@ -0,0 +1,114 @@
+; Test checks "kernel" splitting mode.
+
+; RUN: llvm-split -sycl-split=kernel -S < %s -o %t.files
+; RUN: FileCheck %s -input-file=%t.files_0.ll --check-prefixes CHECK-MODULE0,CHECK
+; RUN: FileCheck %s -input-file=%t.files_0.sym --check-prefixes CHECK-MODULE0-TXT
+; RUN: FileCheck %s -input-file=%t.files_1.ll --check-prefixes CHECK-MODULE1,CHECK
+; RUN: FileCheck %s -input-file=%t.files_1.sym --check-prefixes CHECK-MODULE1-TXT
+; RUN: FileCheck %s -input-file=%t.files_2.ll --check-prefixes CHECK-MODULE2,CHECK
+; RUN: FileCheck %s -input-file=%t.files_2.sym --check-prefixes CHECK-MODULE2-TXT
+
+;CHECK-MODULE2-NOT: @GV
+;CHECK-MODULE1-NOT: @GV
+;CHECK-MODULE0: @GV = internal addrspace(1) constant [1 x i32] [i32 42], align 4
+@GV = internal addrspace(1) constant [1 x i32] [i32 42], align 4
+
+; CHECK-MODULE2: define dso_local spir_kernel void @TU0_kernel0
+; CHECK-MODULE2-TXT: TU0_kernel0
+; CHECK-MODULE1-NOT: define dso_local spir_kernel void @TU0_kernel0
+; CHECK-MODULE1-TXT-NOT: TU0_kernel0
```

Comment:
Shouldn't we add `CHECK-MODULE0-NOT` and `CHECK-MODULE0-TXT-NOT` for @TU0_kernel0?

BTW, I would sort the checks in ascending order (as the run lines).

Reply:
Added missing checks.